src/v1/models/ansible.js

```diff
@@ -115,6 +115,55 @@ export default class AnsibleModel extends KubeModel {
     }
   }
 
+  // clean up all ansible secrets under a namespace if no policy automations
+  async cleanAnsibleSecret(args) {
+    const { namespace } = args;
+    let ansibleSecretDeletion = [];
+
+    // get all policy automations under this namespace
+    const policyAutomations = this.getPolicyAutomations(namespace);
+    // if there isn't any policy automations
+    if (!policyAutomations || Object.entries(policyAutomations).length === 0) {
+      // get all ansible secret under this namespace
+      const ansibleSecrets = await this.kubeConnector.get(
+        `/api/v1/namespaces/${namespace}/secrets?labelSelector=cluster.open-cluster-management.io/type=ans`,
+      );
+      const ansibleSecretItems = ansibleSecrets.items;
+      if (Array.isArray(ansibleSecretItems) && ansibleSecretItems.length > 0) {
+        //  delete ansible secret under this namespace one by one
+        ansibleSecretDeletion = await Promise.all(ansibleSecretItems.map(async (ansibleSecret) => {
+          const ansibleSecretName = _.get(ansibleSecret, 'metadata.name');
+          const results = await this.kubeConnector.delete(`/api/v1/namespaces/${namespace}/secrets/${ansibleSecretName}`);
+          const deletionResult = {
+            name: ansibleSecretName,
+            namespace,
+            deletionStatus: 'N/A',
+          };
+          const { items } = results;
+          if (Array.isArray(items) && items.length > 0) {
```

Comment:
Since the deletion is calling a delete on a particular secret, wouldn't `results` be the deleted object, not an array?

Reply:
It is an always one-item array, length =1

Reply:
Would you please double check this? For `policyAutomationAction`, we're calling `get` for all `policyAutomations` in the namespace, so an `.items` array is returned. For this function, we're calling a specific object, so I think an object is returned. For instance in the tests `mockDeletePolicyResponse` is an object and not an array, so I'm not sure this would be different?